src/System.Management.Automation/engine/PSVersionInfo.cs

```diff
@@ -327,15 +334,47 @@ internal PSVersionHashTable(IEqualityComparer equalityComparer) : base(equalityC
 
         /// <summary>
         /// Returns ordered collection with Keys of 'PSVersionHashTable'
+        /// We want see special order:
+        ///     1. PSVersionName
+        ///     2. PSEditionName
+        ///     3. Remaining properties in alphabetical order
         /// </summary>
         public override ICollection Keys
         {
             get
             {
-                Array arr = new string[base.Keys.Count];
-                base.Keys.CopyTo(arr, 0);
-                Array.Sort(arr, StringComparer.OrdinalIgnoreCase);
-                return arr;
+                ArrayList keyList = new ArrayList(base.Keys);
+                keyList.Sort(new PSVersionTableComparer());
+                return keyList;
+            }
+        }
+
+        private class PSVersionTableComparer : Comparer<object>
+        {
+            public override int Compare(object x, object y)
+            {
+                string xString = (string)LanguagePrimitives.ConvertTo(x, typeof(string), null);
+                string yString = (string)LanguagePrimitives.ConvertTo(y, typeof(string), null);
```

Comment:
What if `x` or `y` are objects but calling `LanguagePrimitives.ConvertTo` on them returns `PSVersion` or `PSEdition` as the converted string?
I still prefer to do sorting only if keys are all strings. If there is a non-string key, we just give up sorting and return `base.Keys`.
Sorry for being picky with rare cases. Since we are creating a public accessible data structure, we have to consider whatever that might happen.